uwblueprint/child-witness-centre
# Email authentication

## Notion Ticket Link

[Email authentication](https://www.notion.so/uwblueprintexecs/Email-Authentication-2e106a8083b44e87bf5846e2c2d06002)



## Implementation Description
* Firebase configured (OAuth Client ID)
* Nodemailer configured (.env)
* Graphql queries for checking user verification



## Steps to Test
1. Add mailer config stuff to your root .env file (check bottom of ticket on Notion 🙈)
2. Re-run docker compose up. :)
3. Go to "Sign Up" and make an account. You should get an email containing a link to verify your account. (don't click it yet!)
4. Try to log in with your credentials. This should not work because you didn't verify your email yet!
5. Now click the link to verify your email!
6. Go try to log in again. This SHOULD work because you did verify your email!



## What should reviewers focus on?
* 


## Screenshots (optional)


## Checklist

## Linting & Formatting
### Mac
```bash
docker exec -it CWC-backend /bin/bash -c "yarn fix"
docker exec -it CWC-frontend /bin/bash -c "yarn fix"
```

### Windows
```bash
docker exec -it CWC-backend bash -c "yarn fix"
docker exec -it CWC-frontend bash -c "yarn fix"
```
